cpp/tests/experimental/indexing/point_quadtree_test.cu

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuspatial/experimental/point_quadtree.cuh>
+#include <cuspatial/vec_2d.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <thrust/copy.h>
+#include <thrust/device_vector.h>
+#include <thrust/fill.h>
+
+namespace {
+
+template <typename T>
+std::vector<T> inline device_uvector_to_host(rmm::device_uvector<T> const& uvec)
+{
+  thrust::device_vector<T> dvec(uvec.size());
+  thrust::fill(dvec.begin(), dvec.end(), T{0});
+  thrust::copy(rmm::exec_policy(uvec.stream()), uvec.begin(), uvec.end(), dvec.begin());
+
+  std::vector<T> hvec(dvec.size());
+  std::fill(hvec.begin(), hvec.end(), T{0});
+  thrust::copy(dvec.begin(), dvec.end(), hvec.begin());
+
+  return hvec;
+}
```

Comment:
For some reason all the other ways I tried copying a `device_uvector` to host kept segfault'ing. I'd love to know if there's a better way to do this.

Reply:
CudaMemcpyAsync, unfortunately.

Reply:
Those fill calls are redundant.

Reply:
Updated